alf/utils/data_buffer.py

```diff
@@ -307,26 +317,58 @@ def _dequeue(self, env_ids=None):
             env_ids = self.check_convert_env_ids(env_ids)
             current_size = self._current_size[env_ids]
             min_size = current_size.min()
-            assert min_size >= 1, (
+            assert min_size >= n, (
                 "Not all environments have enough data. The smallest data "
                 "size is: %s Try storing more data before calling dequeue" %
                 min_size)
             pos = self._current_pos[env_ids] - current_size
-            pos = pos % self._max_length
-            indices = env_ids * self._max_length + pos  # shape [B]
 
-            batch_size = env_ids.shape[0]
-            result = alf.nest.map_structure(
-                lambda buffer: buffer[indices].reshape(
-                    batch_size, 1, *buffer.shape[1:]), self._flattened_buffer)
+            def _create_empty(buf):
+                return []
+
+            result_arr = alf.nest.map_structure(_create_empty, self._buffer)
+
+            for _ in range(n):
```

Comment:
better to not to use loop. It is slow. It should be possible to do all n steps at the same time.

Reply:
Good point.  I tried before without success, but finally found out __getitem__((x,y)) can do the trick.

Btw, I'm not sure why flatten_buffer is used to store and retrieve items from buffer, in this change I've changed to use _buffer instead.  Seems more straightforward.

Let me know if this is Ok.

Reply:
Nice! I didn't know that `__getitem__()` can take tuple. That's why flattened_buffer is used for ReplayBuffer.get_batch(). Can you also change that to use `__getitem__()`?